internals/src/macros.rs

```diff
@@ -199,3 +199,17 @@ macro_rules! impl_from_infallible {
         }
     }
 }
+
+/// Implements `to_hex` for functions that have implemented [`core::fmt::LowerHex`]
+#[macro_export]
+macro_rules! impl_to_hex_from_lower_hex {
+    ($t:ident) => {
+        impl $t {
+            /// Gets the hex representation of this type
+            #[cfg(feature = "alloc")]
```

Comment:
The feature is never enabled because it is applied to the calling crate, and `bitcoin` does not have an `alloc` feature.

Reply:
Does it make sense to add an `alloc` feature into `bitcoin`?

Reply:
I don't think we should do so, an allocator is a hard requirment ATM in the `bitcoin` crate.

Reply:
It's possible but very laborious, I'd rather fix the macro.